go-cloudfs/main.go

```diff
@@ -36,34 +36,55 @@ func main() {
 	r := chi.NewRouter()
 	r.Use(middleware.Logger)
 
-	r.Post("/write-file", func(writer http.ResponseWriter, r *http.Request) {
-		w, err := bucket.NewWriter(r.Context(), "foo.txt", nil)
+	r.Post("/write-file/{path}", func(writer http.ResponseWriter, req *http.Request) {
+		path := chi.URLParam(req, "path")
+		fmt.Println(path)
+		w, err := bucket.NewWriter(req.Context(), path, nil)
 		if err != nil {
+			log.Fatal(err)
```

Comment:
I just discovered yesterday that `log.Fatal` always results in `os.Exit(1)`, and you can't override it.

So, all of these `log.Fatal`s should probably be `log.Error`.

Sorry, I forgot that in my previous review.

Reply:
Ah ok let me change

Reply:
there is no log.Error it seems?

Reply:
Oh, I'm sorry — I was thinking the zap logger. I have no idea how this log works. Probably fine with Fatal.

Sorry for the wild goose chase :( None of this is blocking.